src/Nethermind/Nethermind.Evm/Tracing/GethStyle/GethLikeTxTracer.cs

```diff
@@ -68,13 +68,14 @@ public abstract class GethLikeTxTracer<TEntry> : GethLikeTxTracer where TEntry :
     protected GethLikeTxTracer(GethTraceOptions options) : base(options) { }
     private bool _gasCostAlreadySetForCurrentOp;
 
-    public override void StartOperation(int depth, long gas, Instruction opcode, int pc, bool isPostMerge = false)
+    public override void StartOperation(in ExecutionEnvironment env, long gas, Instruction opcode, int pc)
     {
+        bool isPostMerge = env.TxExecutionContext.BlockExecutionContext.Header.IsPostMerge;
```

Comment:
Maybe have simple `IsPostMerge` extension method on `ExecutionEnvironment` to simplify nesting